dial_back_tracker.go

```diff
@@ -0,0 +1,27 @@
+package autonat
+
+import (
+	"github.com/libp2p/go-libp2p-core/network"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/patrickmn/go-cache"
+	"time"
+)
+
+var _ network.Notifiee = (*dialBackTracker)(nil)
+
+type dialBackTracker client
+
+func (d *dialBackTracker) Listen(net network.Network, a ma.Multiaddr)         {}
+func (d *dialBackTracker) ListenClose(net network.Network, a ma.Multiaddr)    {}
+func (d *dialBackTracker) OpenedStream(net network.Network, s network.Stream) {}
+func (d *dialBackTracker) ClosedStream(net network.Network, s network.Stream) {}
+
+func (d *dialBackTracker) Connected(net network.Network, c network.Conn) {
+	dialer := c.RemotePeer()
```

Comment:
@Stebalien Since this & the dial back response happen on separate go-routines, we could see the dial back response before this code has had a chance to execute. However, the possibility of that seems low to me given the RTT times. Should we handle this race or just document it ?

Reply:
This'll be a problem for QUIC because the initiator will learn that the connection has finished before the receiver. That is, the initiator will receive the second message in the three-way handshake, send the final message, then start sending data.

Assuming the initiator (the AutoNAT server) acts fast enough, they can send the dial back response at pretty much the same time as the final handshake packet. At that point, it's a matter of scheduling and/or lost packets.

Reply:
But let's not put too much work into this before we discuss it a bit.